### ISSUE TITLE
"Find in files" and "Open recent" fail on *.y files

### ISSUE THREAD

**Issue by dmurdoch** [rstudio/rstudio#10741]
### System details

    RStudio Edition : Desktop
    RStudio Version : "Prairie Trillium" Release (9f796939, 2022-02-16) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.10 Chrome/69.0.3497.128 Safari/537.36
    R Version       : 4.1.0, but probably not relevant

### Steps to reproduce the problem

Create a file named `sample.y` with this content:

```
This is some text in a file
named sample.y.

If I do an RStudio "Find in Files"
search for the string "sample.y"
I see two hits in this file.

However, if I click on those hits,
the editor won't jump to the
listed lines.

```

(The .y extension is used in Bison/yacc files; there are a couple of them in the R sources, which is where I noticed this.)

Do a "Find in files" search through all files, or a custom filter selecting *.y files, for the string "sample.y".

Click on one of the hits.  Nothing happens.

Close the file, and then go to the `File | Recent files` list, and the file should appear there, but clickin on it won't open it.

### Describe the problem in detail

The search finds the two occurences of that string in the `sample.y` file, and correctly lists the lines.  But if you click on those two hits, nothing happens.  If the file is open in the editor, the cursor doesn't move to the right line.  If the file is not open, the editor doesn't open it.  The `File | Recent files` entry shows up, but doesn't work.  

### Describe the behavior you expected

For most searches, clicking on a hit makes the editor jump to the corresponding line.  Choosing a recent file opens it.

- [ x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Thanks for the bug report! I've reproduced this locally as well; it seems to affect RStudio Desktop but not RStudio Server.

The issue appears to lie here:

https://github.com/rstudio/rstudio/blob/f8c4fea1bb16acfc2a70b5feacdd5838b6f9ebaa/src/gwt/src/org/rstudio/studio/client/common/filetypes/FileTypeRegistry.java#L541-L571

Because we don't have a registered filetype for files with the `.y` extension, this code fails to infer the file type and so fails to open the file.

I think we should have a fallback for cases like this; if we don't have a registered filetype, but the file appears to be a text file, we should still allow users to open it.

**Comment by MariaSemple** [comment]
Does this occur when attempting to open similar files (ones that aren't registered with an file type) through other menus, such as the Files pane or the Open Files dialog?

**Comment by dmurdoch** [comment]
Files pane:  no problem, opened as text file.

File | Open file... dialog:  no problem, opened as text file.

In Finder, using Open with.. and choosing RStudio:   no problem, opened as text file.

**Comment by romainfrancois** [comment]
Verified in RStudio 2022.06.0-daily+406 "Spotted Wakerobin" Daily

**Comment by dmurdoch** [comment]
I just updated RStudio after being prompted to do so, and this isn't fixed there.  That may be because 2022.06.0-daily+406 contains things that haven't made it to the release yet, but just in case this was intended to be fixed in what I just installed, I'm reporting it here.  This is my new version info:

```
RStudio 2022.02.3+492 "Prairie Trillium" Release (1db809b8323ba0a87c148d16eb84efe39a8e7785, 2022-05-16) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.10 Chrome/69.0.3497.128 Safari/537.36
```

**Comment by jgutman** [comment]
You've updated RStudio, but not to the 2022.06.0 versions, which have not yet been released, they are only available in dailies. You can download it from dailies.rstudio.com if you would like to try it out, otherwise it should be officially released likely sometime next month. The fix is not available in the version you just downloaded, 2022.02.3.
